compiler/optimizations/deadCodeElimination.cpp

```diff
@@ -437,6 +464,16 @@ static void deadBlockElimination(FnSymbol* fn)
         // then remove the entire While.
         whileStmt->remove();
 
+      else if (isSymExpr(expr))
```

Comment:
@noakesmichael : Apparently by the time deadCodeElimination is run as an optimization, ForLoop statements have been removed from the tree.  Otherwise, you would have encountered this problem in your development branch.
Basic block analysis is so generic, it ought to be possible to run it at any point in the translation.  As a start, it would be reasonable to add 
  BasicBlock::buildBasicBlocks(fn)
as a verify test and expect this to run correctly after every pass.